packages/graphile-build-pg/src/plugins/PgTypesPlugin.js

```diff
@@ -341,6 +341,14 @@ export default (function PgTypesPlugin(
         inflection.builtin("InternetAddress"),
         "An IPv4 or IPv6 host address, and optionally its subnet."
       );
+      const CidrType = stringType(
+        inflection.builtin("CidrAddress"),
+        "An IPv4 or IPv6 CIDR address."
+      );
+      const MacAddrType = stringType(
+        inflection.builtin("MacAddress"),
+        "A 6-byte MAC address."
+      );
```

Comment:
I believe that cidr and macaddr currently use the `String` type; if you could make this an opt-in (non-breaking) change I'd be happy to merge it. To do so we need to do a couple of small changes:

First, this section needs to use a new flag `pgUseCustomNetworkScalars`. Add the new flag (`pgUseCustomNetworkScalars = false`) here:
https://github.com/graphile/graphile-engine/blob/e1d167e30f725707483fbea66b9bed28b80c5caf/packages/graphile-build-pg/src/plugins/PgTypesPlugin.js#L33

Secondly, change these two defs to use the new flag: `const CidrType = pgUseCustomNetworkScalars ? stringType(...) : GraphQLString`

This would make this available using `graphileBuildOptions`. I don't think we should expand it beyond here as we'll just default it to on in V5.

Reply:
Updated the PR with the requested changes, macaddr8 type (pg10+), and rebased on top of master. I had to make a few design decisions in the test suite to test the various schema permutations with the default option, opted-in option, and pg10 specific tests (mutations and queries).

Let me know if there is anything I can improve to get this pulled. Thank you for your feedback.